### PR TITLE
Set 600 mode file permissions on temp KUBECONFIG

### LINKED ISSUE #14: [Security] Temporary KUBECONFIG world readable
Please force permission on temporary KUBECONFIG to be not group/world readable, this should not follow system/user umask.

```
$ umask
0022
$ kubie ctx default
$ echo $KUBECONFIG
/tmp/kubie-config1585937564032423433.yaml
$ ls -la /tmp/kubie-config1585937564032423433.yaml
-rw-r--r-- 1 user group 1109 Apr  3 20:12 /tmp/kubie-config1585937564032423433.yaml
```

### PR DESCRIPTION
**Pull request by TyBrown**
Fixes https://github.com/sbstp/kubie/issues/14#issuecomment-1404654423

### CODE REVIEW

File: src/kubeconfig.rs [hunk 1]
```diff
@@ -72,8 +72,10 @@ pub struct Installed {
 }
 
 impl KubeConfig {
-    pub fn write_to<W: Write>(&self, writer: W) -> anyhow::Result<()> {
-        serde_yaml::to_writer(writer, self)?;
+    pub fn write_to_file(&self, path: &Path) -> anyhow::Result<()> {
+        let file = File::create(path).context("could not write file")?;
+        fs::set_permissions(&path, Permissions::from_mode(0o600))?;
+        serde_yaml::to_writer(file, self)?;
```
Comment (sbstp) [inline src/kubeconfig.rs]: LGTM overall. Only concern is that there's no buffering on top of the `File`. `serde` makes a bunch of `write` calls with small buffers that will all end up being syscalls if no buffering is put in place.

Comment (TyBrown) [reply src/kubeconfig.rs]: I think I've addressed this comment by using a `BufWriter` here... let me know please if I misunderstood. :)

Comment (sbstp) [reply src/kubeconfig.rs]: Yep that's exactly what I had in mind, thanks.

**Comment by sbstp** [comment]
It's out in 0.19.2

**Pull request merged**